baseapp/abci.go

```diff
@@ -833,10 +833,6 @@ func (app *BaseApp) internalFinalizeBlock(ctx context.Context, req *abci.Request
 		txResults = append(txResults, response)
 	}
 
-	if app.finalizeBlockState.ms.TracingEnabled() {
-		app.finalizeBlockState.ms = app.finalizeBlockState.ms.SetTracingContext(nil).(storetypes.CacheMultiStore)
```

Comment:
With a second thought, I guess the intension here is probably to reset the txHash to nil?

Reply:
interesting, the issue lies in [`cachemulti.Store`](https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/store/cachemulti/store.go#L97), it's not a reset there.